packages/beacon-state-transition/src/allForks/util/cachedBeaconState.ts

```diff
@@ -253,3 +276,13 @@ export const CachedBeaconStateProxyHandler: ProxyHandler<CachedBeaconState<allFo
     return false;
   },
 };
+
+/**
+ * Caution: not for production, only used for sim tests.
+ */
+export function clearDomainCache(): void {
+  domainCache.clear();
+}
+
+/** Domain cache, share across beacon states */
+const domainCache = new Map<ForkName, Map<DomainType, Buffer>>();
```

Comment:
Having non-universal data attached to the global scope is a bad idea. If someone runs two nodes with different genesis it's can break everything. Also it makes it harder to track memory and not good etiquette anyway. We should only do it for very specific occasions like caching hex to uint conversions and uint to boolean[].